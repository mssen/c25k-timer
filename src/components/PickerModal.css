.overlay, .modal-container {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.overlay {
  z-index: 50;
  background: rgba(0, 0, 0, 0.6);
}

.modal-container {
  z-index: 75;
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal {
  background: white;
  width: 650px;
  max-width: 100%;
  height: 500px;
  max-height: 100%;
  padding: 10px 20px;
}

.modal-header {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.modal-body {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.close-button {
  cursor: pointer;
  background: transparent;
  border: 0;
  font-size: 25px;
}

.picker-button {
  cursor: pointer;
  border-radius: 4px;
  border: 0;
  font-size: 16px;
  text-decoration: underline;
  background: transparent;
  padding: 10px 20px;
  margin: 8px 5px;
}

.picker-button:hover {
  background: lightgray;
}