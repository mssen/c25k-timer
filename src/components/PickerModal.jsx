import React from 'react';
import PropTypes from 'prop-types';

import './PickerModal.css';

const PickerModal = ({ segments, currentSegment, hideModal, updateSegment }) => (
  <div>
    <div className="overlay" />
    <div className="modal-container">
      <div className="modal">
        <header className="modal-header">
          <h1>Pick a Day</h1>

          <button className="close-button" onClick={hideModal}>
            âœ•
          </button>
        </header>

        <div className="modal-body">
          {segments.map((segment, index) => (
            <button
              className="picker-button"
              key={index}
              onClick={() => updateSegment(index - currentSegment)}
            >
              {`Week ${segment.week}, Day ${segment.day}`}
            </button>
          ))}
        </div>
      </div>
    </div>
  </div>
);

PickerModal.propTypes = {
  segments: PropTypes.array.isRequired,
  currentSegment: PropTypes.number.isRequired,
  hideModal: PropTypes.func.isRequired,
  updateSegment: PropTypes.func.isRequired
};

export default PickerModal;
