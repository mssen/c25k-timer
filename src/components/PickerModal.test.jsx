import React from 'react';
import { shallow } from 'enzyme';
import PickerModal from './PickerModal';

const segments = [
  { week: 1, day: 1 },
  { week: 1, day: 2 },
  { week: 1, day: 3 },
  { week: 2, day: 1 },
  { week: 2, day: 2 },
  { week: 2, day: 3 }
];

it('renders without crashing', () => {
  shallow(
    <PickerModal
      segments={[]}
      currentSegment={0}
      hideModal={() => ''}
      updateSegment={() => ''}
    />
  );
});

it('renders weeks and days', () => {
  const wrapper = shallow(
    <PickerModal
      segments={segments}
      currentSegment={0}
      hideModal={() => ''}
      updateSegment={() => ''}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('handles hiding the modal', () => {
  const hideModalStub = jest.fn();
  const wrapper = shallow(
    <PickerModal
      segments={[]}
      currentSegment={0}
      hideModal={hideModalStub}
      updateSegment={() => ''}
    />
  );
  wrapper
    .find('button')
    .first()
    .simulate('click');
  expect(hideModalStub).toBeCalled();
});

it('handles updating the segment', () => {
  const updateSegmentStub = jest.fn();
  const wrapper = shallow(
    <PickerModal
      segments={segments}
      currentSegment={3}
      hideModal={() => ''}
      updateSegment={updateSegmentStub}
    />
  );
  wrapper
    .find('div > button')
    .first()
    .simulate('click');
  expect(updateSegmentStub).toBeCalledWith(-3);
});
