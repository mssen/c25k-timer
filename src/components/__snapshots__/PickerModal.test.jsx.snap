// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders weeks and days 1`] = `
<div>
  <header>
    <h1>
      Pick a Day
    </h1>
    <button>
      Close
    </button>
  </header>
  <button
    onClick={[Function]}
  >
    Week 1, Day 1
  </button>
  <button
    onClick={[Function]}
  >
    Week 1, Day 2
  </button>
  <button
    onClick={[Function]}
  >
    Week 1, Day 3
  </button>
  <button
    onClick={[Function]}
  >
    Week 2, Day 1
  </button>
  <button
    onClick={[Function]}
  >
    Week 2, Day 2
  </button>
  <button
    onClick={[Function]}
  >
    Week 2, Day 3
  </button>
</div>
`;
